test/users/user_boundaries_test.exs

```diff
@@ -0,0 +1,115 @@
+  alias Bonfire.Boundaries.Grants
+  alias Bonfire.Me.Fake
+  alias Bonfire.Me.Users
+  alias Bonfire.Boundaries.Users
+  alias Bonfire.Common.Config
+  alias Bonfire.Boundaries.Circles
+  alias Bonfire.Boundaries.Acls
+
+  describe "default boundaries from config should be inserted in the database when a new user is created" do
+    setup do
+      on_exit(fn -> Process.delete([:bonfire, :user_default_boundaries]) end)
+    end
+
+    test "nothing should be created if the configs are empty" do
+      Process.put([:bonfire, :user_default_boundaries], %{
+        circles: %{},
+        acls: %{},
+        grants: %{},
+        controlleds: %{}
+      })
+
+      %{id: user_id} = user = fake_user!()
+      assert length(Circles.list_my(user)) == 0
+      assert repo().one(from g in Grant, select: count(g), where: g.subject_id == ^user_id) == 0
+    end
+
+    test "circles should be created" do
+      Process.put([:bonfire, :user_default_boundaries], %{
+        circles: %{
+          test_circle: %{stereotype: :followers}
+        },
+        acls: %{},
+        grants: %{},
+        controlleds: %{}
+      })
+
+      %{id: user_id} = user = fake_user!()
+      assert length(Circles.list_my(user)) == 1
+    end
+
+    test "grants should be created" do
+      Process.put([:bonfire, :user_default_boundaries], %{
+        circles: %{},
+        acls: %{},
+        grants: %{
+          i_may_administer: %{
+            SELF: [:see, :read]
+          }
+        },
+        controlleds: %{}
+      })
+
+      %{id: user_id} = user = fake_user!()
+      assert repo().one(from g in Grant, select: count(g), where: g.subject_id == ^user_id) == 2
+    end
+
+    test "acls should be created" do
+      Process.put([:bonfire, :user_default_boundaries], %{
+        circles: %{},
+        acls: %{i_may_administer: %{stereotype: :i_may_administer}},
+        grants: %{
+          i_may_administer: %{
+            SELF: [:see, :read]
+          }
+        },
+        controlleds: %{}
+      })
+
+      %{id: user_id} = user = fake_user!()
+
+      assert length(Acls.list_my(user, paginate?: false)) == 1
+    end
+
+    test "default visibility controlleds should be created" do
+      Process.put([:bonfire, :user_default_boundaries], %{
+        circles: %{},
+        acls: %{},
+        grants: %{ },
+        controlleds: %{SELF: []}
+      })
+
+      %{id: user_id} = user = fake_user!()
+      assert repo().one(from c in Controlled, select: count(c), where: c.id == ^user_id) == 1
+       end
+
+       test "controlleds should be created" do
+        Process.put([:bonfire, :user_default_boundaries], %{
+          circles: %{},
+          acls: %{},
+          grants: %{ },
+          controlleds: %{SELF: [
+          :locals_may_reply,
+          :remotes_may_reply,
+          :i_may_administer,]}
+        })
+
+        %{id: user_id} = user = fake_user!()
+        assert repo().one(from c in Controlled, select: count(c), where: c.id == ^user_id) == 4
```

Comment:
and here it's granted others (local and remote users) permission to reply/etc, but is `i_may_administer` a dup of what's already being applied by default?

Reply:
I'm not super sure about the semantics. I see 1 entry with nothing in the config and 1+3 entries with 3 entries in the config, so I was happy with that lol.

Reply:
yeah that's fair, I'll run the tests and inspect the result, as its a good opportunity to document the behaviour